### PR TITLE
Update proxy info on logs

### PR DESCRIPTION
**Pull request by NBParis**
### What does this PR do?
Explain how to use a third party proxy with logs

### Motivation
There was confusion about what could be done with proxies and logs.

### Preview link
https://docs-staging.datadoghq.com/nils/proxy-logs/logs/log_collection/

### CODE REVIEW

File: content/logs/log_collection/_index.md [hunk 1]
```diff
@@ -237,7 +237,18 @@ logs:
 
 ### Using a Proxy for Logs
 
-The log Agent does not presently respect the proxy setting in the datadog.yaml configuration file. This feature will be available in a future release.
+The log Agent does not presently respect the usual proxy setting in the `datadog.yaml` configuration file. It is **not yet** possible to use the agent as proxy (local agent -> proxy agent -> Datadog) but a third party proxy can be used.
```
[the remark this answers is not in the document]
Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: "local agent -> proxy agent -> Datadog" is not supported for any version of the agent at this point. 
Can we just link this to the main proxy docs instead? We need to honor the same proxy settings across all agents.

File: content/logs/log_collection/_index.md [hunk 2]
```diff
@@ -237,7 +237,18 @@ logs:
...
+
+To use a third party proxy, configure your Datadog Agents to forward logs to this proxy by setting those parameters in their respective `datadog.yaml` configuration file:
```
[the remark this answers is not in the document]
Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: This seems no different than https://docs.datadoghq.com/agent/proxy/

File: content/logs/log_collection/_index.md [hunk 3]
```diff
@@ -237,7 +237,18 @@ logs:
...
+
+```
+logs_config:
+  dd_url: <MY_PROXY_URL>
+  dd_port: <MY_PROXY_PORT>
+  dev_mode_no_ssl: true
+ ```
+
+Then configure your proxy to forward logs to the endpoint `agent-intake.logs.datadoghq.com` on port `10516` with SSL activated.
```
[the remark this answers is not in the document]
Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: Theres no reason not to suggest proxies like we do here ad provide example configs for things like haproxy, 
https://docs.datadoghq.com/agent/proxy/

[the remark this answers is not in the document]
Comment (l0k0ms) [reply content/logs/log_collection/_index.md]: Haproxy works and is recommanded, Squid doesn't support manual configuration for TCP for instance.

File: content/logs/log_collection/_index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (l0k0ms) [reply content/logs/log_collection/_index.md]: I updated the text in order to explain why there is a difference between log traffic and metric traffic.

File: content/logs/log_collection/_index.md [hunk 4]
```diff
@@ -237,19 +237,21 @@ logs:
 
 ### Using a Proxy for Logs
 
-The log Agent does not presently respect the usual proxy setting in the `datadog.yaml` configuration file. It is **not yet** possible to use the agent as proxy (local agent -> proxy agent -> Datadog) but a third party proxy can be used.
+The log Agent does not presently respect the usual proxy setting in the `datadog.yaml` configuration file, since logs are only forwarded through TCP/SSL and not HTTPs.
```
Comment (irabinovitch) [inline content/logs/log_collection/_index.md]: - consistent caps please.  "log Agent"
- Also Datadog has one Datadog Agent. Logs, Traces, Metrics are Processes are features it provides. Lets avoid language that implies theres a separate log agent package.

Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: "Logs does does not presently respect the global proxy settings in the `datadog.yaml` configuration file. This is due to logs presently being transported over TCP/SSL while other Datadog features rely on HTTPS."

OR better yet:

"Logs make use of a different set of proxy settings than other data types forwarded by the Datadog Agent.  This is due to logs presently being transported over TCP/SSL, while other features submit data via on HTTPS."


Do we need to call out environment variables in addition to datadog.yaml?

File: content/logs/log_collection/_index.md [hunk 5]
```diff
@@ -237,19 +237,21 @@ logs:
...
 
-To use a third party proxy, configure your Datadog Agents to forward logs to this proxy by setting those parameters in their respective `datadog.yaml` configuration file:
+Configure your Datadog Agent to forward logs to a proxy by setting those parameters in your Agent  `datadog.yaml` configuration file:
```
Comment (irabinovitch) [inline content/logs/log_collection/_index.md]: s/those/these/

Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: "To configure your Datadog Agent to forward logs through a proxy server, add these settings to the `datadog.yaml` configuration file:"

File: content/logs/log_collection/_index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (irabinovitch) [reply content/logs/log_collection/_index.md]: OK. Haproxy is fine.

**Comment by NBParis** [comment]
@irabinovitch Let us know if the latest modification works for you so we can merge it.

**Comment by irabinovitch** [comment]
@l0k0ms  @NBParis Merging but i think this page should eventually go away and be merged into the main proxy docs.  It's all one agent from our customers perspective. These are just new settings / config types.

**Comment by l0k0ms** [comment]
As soon as the HTTP endpoint is released for logs I will merge this doc into the main proxy documentation.

**Pull request merged**